docs/rules.rst

```diff
@@ -0,0 +1,622 @@
+  entry is found, its weight will be used for that rule, if not, the
+  ``item.weight`` specified in the rule will be used instead.
+
+``when``
+  Optional. Define condition for the rule to be enabled.
+
+.. _ferm__rules: defaults.html#envvar-ferm__rules
+.. _ferm__group_rules: defaults.html#envvar-ferm__group_rules
+.. _ferm__host_rules: defaults.html#envvar-ferm__host_rules
+.. _ferm__dependent_rules: defaults.html#envvar-ferm__dependent-rules
+.. _ferm__domains: defaults.html#envvar-ferm__domains
+.. _ferm__weight_map: defaults.html#envvar-ferm__weight_map
+
+.. _rule_templates:
+
+Depending on the choosen type, many additional variables are supported.
+Please check the individual template description below.
+
+
+Rule templates
+--------------
+
+There exist a number of predefined rule templates for generating firewall
+rules through ferm. Each rule definition is referencing the used template
+through its ``type`` key. The templates are located in the
+``templates/etc/ferm/ferm.d/`` directory.
+
+Following a list of the available rule templates which can be used to
+create custom rules:
+
+* ``accept``: Template to create rules that match interfaces, ports, remote
+  IP addresses/subnets and can accept the packets, reject, or redirect to
+  a different chain. The following template-specific YAML keys are supported:
+
+    ``accept_any``
+      Optional. ?
+
+    ``daddr``
+      Optional. List of destination IP addresses or networks to which the
+      rule is applied.
+
+    ``dport``
+      Optional. List of destination ports to which the rule is applied.
+
+    ``enabled``
+      Optional. Enable rule definition. Possible values: ``True`` or
+      ``False``. Defaults to ``True``.
+
+    ``include``
+      Optional. Custom ferm configuration file to include.
+
+    ``interface``
+      Optional. List of network interfaces for incoming packets to which
+      the rule is applied.
+
+    ``interface_present``
+      Optional. Same as ``item.interface`` but first check if specified
+      network interfaces exists before adding the firewall rules.
+
+    ``multiport``
+      Optional. Use ``iptables multiport`` extension. Possible values:
+      ``True`` or ``False``. Defaults to ``False``.
+
+    ``name``
+      Optional. Set rule name in places where it is referenced.
+
+    ``outerface``
+      Optional. List of network interfaces for outgoing packets to which
+      the rule is applied.
+
+    ``outerface_present``
+      Optional. Same as ``item.outerface`` but first check if specified
+      network interface exists before adding the firewall rule.
+
+    ``protocol``
+      Optional. Network protocol to which the rule is applied.
+
+    ``protocol_syn``
+      Optional. Match TCP packet with only the SYN flag set. Possible
+      values ``True`` or ``False``. If set to ``False`` it will match all
+      other packets except the ones with only the SYN flag set. Defaults
+      to unset.
+
+    ``realgoto``
+      Optional. ?
```

Comment:
This is an internal `ferm` notation used to change the `JUMP` behaviour: http://ferm.foo-projects.org/download/2.1/ferm.html#basic_target_keywords